### PR TITLE
[shell] Add open-ssh to shell service

### PR DESCRIPTION
**Pull request by sahare92**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: stable/shell/templates/shell-configmap.yaml [hunk 1]
```diff
@@ -146,4 +146,12 @@ data:
       echo "# ----- END KUBECTL AUTOCOMPLETION -----" >> ${HOME}/.bashrc
     fi
 
+{{- if .Values.sshServer.enabled }}
+    cp /var/run/iguazio/secrets/ssh/* /igz/.ssh
+    chown iguazio:iguazio /igz/.ssh/*
+    chmod 0644 /igz/.ssh/*
```
Comment (omesser) [inline stable/shell/templates/shell-configmap.yaml]:
```suggestion
    chmod -R 0644 /igz/.ssh/
```

File: stable/shell/templates/shell-configmap.yaml [hunk 2]
```diff
@@ -146,4 +146,12 @@ data:
       echo "# ----- END KUBECTL AUTOCOMPLETION -----" >> ${HOME}/.bashrc
     fi
 
+{{- if .Values.sshServer.enabled }}
+    cp /var/run/iguazio/secrets/ssh/* /igz/.ssh
+    chown iguazio:iguazio /igz/.ssh/*
```
Comment (omesser) [inline stable/shell/templates/shell-configmap.yaml]:
```suggestion
    chown -R iguazio:iguazio ${HOME}/.ssh/
```
How does this work with `/igz/.ssh` :\

File: stable/shell/values.yaml [hunk 3]
```diff
@@ -58,6 +58,15 @@ daemon:
     tag: "1.0.0"
     pullPolicy: "IfNotPresent"
 
+sshServer:
+  enabled: false
+
+  # can specifically set the port for the service NodePort (optional)
+  # port:
```
Comment (omesser) [inline stable/shell/values.yaml]:
```suggestion
  # nodePort:
```
Also please add to the comment that if one is not given, a random one will be assigned

File: stable/shell/values.yaml [hunk 4]
```diff
@@ -58,6 +58,15 @@ daemon:
...
+
+  # base64 encoded public key
+  publicKey: ""
```
Comment (omesser) [inline stable/shell/values.yaml]: `authorized_keys` file can contain multiple keys.
So either name that `authorizedKeys` and have it accept a base64 encoding of the entire file, or name it `publicKeys` which will be a list of base64 encoded strings , each of 1 publicKey

**Review by omesser — CHANGES_REQUESTED** [review]

File: stable/shell/templates/shell-configmap.yaml (hunk 2 above)
Comment (urihoenig) [reply stable/shell/templates/shell-configmap.yaml]: Why do you chown here? Who is the original owner?
I'm asking as we are about to use other users to execute jupyter (and I guess that shell will follow)

Comment (urihoenig) [reply stable/shell/templates/shell-configmap.yaml]: And why /igz/.ssh? This is not the users's home dir
The user's home path is ${HOME} as Oded indicated.
If you use it, you have to keep in mind that this is a persistent path on our fuse (I don't know whether you want to override it or not).
If it's there, there's no point in chowning it at all.
Do you want to allow the user to edit those files?
Is the secret here just a way to initialize it?

**Review by urihoenig — CHANGES_REQUESTED** [review]

File: stable/shell/templates/shell-configmap.yaml (hunk 2 above)
Comment (sahare92) [reply stable/shell/templates/shell-configmap.yaml]: - using $HOME doesn't work - as we don't have users other than _iguazio_.
and the _iguazio_ user's home dir is `/igz` (which is $IGUAZIO_HOME)
- and true - it seems there's no need for chown here
- about editing the public keys, I'm not sure if there's a need/restriction for that - I'll ask product to make sure

Comment (urihoenig) [reply stable/shell/templates/shell-configmap.yaml]: $HOME should point you to /User which isn't iguazio's home, but the users's fuse home (could be any user).
We manage all the user's stuff there (.bashrc etc.).
Not sure what's the right way to go here, but if you don't use it, there's no persistence anyway (no way to edit the keys outside the secret).

**Review by urihoenig — APPROVED** [review]

**Pull request merged**